tests/reporter/kubernetes.go

```diff
@@ -63,35 +62,36 @@ const (
 	networkAttachmentDefinitionEntity = "network-attachment-definitions"
 )
 
+type KubernetesReporterOptions struct {
+	ArtifactsDir     string
+	MaxFails         int
+	Namespaces       []string
+	IsRunningOnKind  bool
+	InstallNamespace string
+	KubectlPath      string
+	OCPath           string
+}
+
 type KubernetesReporter struct {
-	failureCount    int
-	artifactsDir    string
-	maxFails        int
-	namespaces      []string
-	isRunningOnKind bool
+	opts         KubernetesReporterOptions
```

Comment:
Would it make sense to embed the struct ? this way, you'd avoid plenty of changes in the client code.